superduperdb/vector_search/server/app.py

```diff
@@ -0,0 +1,117 @@
+
+from superduperdb.base.datalayer import Datalayer
+from superduperdb.vector_search.server import service
+
+app = FastAPI()
+
+
+class VectorItem(BaseModel):
+    id: str
+    vector: t.List
+
+
+# --------------- Create datalayer on start -----------------
+
+
+@app.on_event('startup')
+def startup_db_client():
+    db = service.create_datalayer()
+    app.state.pool = db
+
+
+@app.on_event('shutdown')
+def shutdown_db_client():
+    try:
+        app.state.pool.close()
+    except AttributeError:
+        raise Exception('Could not close the database properly')
+
+
+# --------------- Create search routes -----------------
+
+
+@app.get("/")
+def home():
+    return {"message": "Vector search server"}
+
+
+@app.get("/create/search", status_code=200)
+def create_search(vector_index: str, db: Datalayer = Depends(service.database)):
+    service.create_search(vector_index=vector_index, db=db)
+    return {'message': 'Vector index created successfuly'}
+
+
+@app.post("/query/id/search")
+def query_search_by_id(
+    id: str, vector_index: str, n: int = 100, db: Datalayer = Depends(service.database)
+):
+    ids, scores = service.query_search_from_id(
+        id, vector_index=vector_index, n=n, db=db
+    )
+    return {'ids': ids, 'scores': scores}
+
+
+@app.post("/query/search")
+def query_search_by_array(
+    vector: service.ListVectorType,
+    vector_index: str,
+    n: int = 100,
+    db: Datalayer = Depends(service.database),
+):
+    ids, scores = service.query_search_from_array(
+        vector, vector_index=vector_index, n=n, db=db
+    )
+    return {'ids': ids, 'scores': scores}
+
+
+@app.post("/add/search")
+def add_search(
+    vectors: t.List[VectorItem],
+    vector_index: str,
+    db: Datalayer = Depends(service.database),
+):
+    service.add_search(vectors, vector_index=vector_index, db=db)
+    return {'message': 'Added vectors successfuly'}
+
+
+@app.post("/delete/search")
+def delete_search(
+    ids: t.List[str], vector_index: str, db: Datalayer = Depends(service.database)
+):
+    service.delete_search(ids, vector_index=vector_index, db=db)
+    return {'message': 'Ids deleted successfuly'}
+
+
+@app.get("/list/search")
+def list_search(db: Datalayer = Depends(service.database)):
+    return service.list_search(db)
+
+
+# --------------- Create exception handler middleware-----------------
+
+
+class ExceptionHandlerMiddleware(BaseHTTPMiddleware):
+    async def dispatch(self, request: Request, call_next):
+        try:
+            return await call_next(request)
+        except Exception as e:
+            return JSONResponse(
+                status_code=500,
+                content={'error': e.__class__.__name__, 'messages': e.args},
+            )
+
+
+app.add_middleware(ExceptionHandlerMiddleware)
+
+
+# --------------- Create vector search serve -----------------
+def serve():
+    # TODO: configurable
+    uvicorn.run(app, host="0.0.0.0", port=8000, reload=False)
```

Comment:
Can we add a configuration?

Reply:
what to add in Config: class

vector_search_service as a block ?

Reply:
`lance://localhost:8000` or `inmemory://localhost:8000` (this at the bottom level)

Reply:
This would make it consistent with the other things, and open to add new configurations.